lib/authorizer/hawk.js

```diff
@@ -210,45 +210,52 @@ module.exports = {
                 'extraData',
                 'app',
                 'delegation',
-                'user'
+                'user',
+                'includePayloadHash'
             ]),
             contentType = request.headers.get('content-type'),
 
             // force toString to add a protocol to the URL.
             url = urlEncoder.toNodeUrl(request.url.toString(true)),
 
-            self = this;
+            self = this,
+
+            signRequest = function (bodyHash) {
+                result = self.computeHeader({
+                    credentials: {
+                        id: params.authId,
+                        key: params.authKey,
+                        algorithm: params.algorithm
+                    },
+                    nonce: params.nonce,
+                    timestamp: params.timestamp,
+                    ext: params.extraData,
+                    app: params.app,
+                    dlg: params.delegation,
+                    user: params.user,
+                    url: url.href, // force toString to add a protocol to the URL.
+                    method: request.method,
+                    hash: bodyHash
+                });
+                request.addHeader({
+                    key: 'Authorization',
+                    value: result,
+                    system: true
+                });
+
+                return done();
+            };
 
         if (!params.authId || !params.authKey) {
             return done(); // Nothing to do if required parameters are not present.
         }
 
         request.removeHeader('Authorization', {ignoreCase: true});
 
-        computeBodyHash(request.body, params.algorithm, 'base64', contentType, function (bodyHash) {
-            result = self.computeHeader({
-                credentials: {
-                    id: params.authId,
-                    key: params.authKey,
-                    algorithm: params.algorithm
-                },
-                nonce: params.nonce,
-                timestamp: params.timestamp,
-                ext: params.extraData,
-                app: params.app,
-                dlg: params.delegation,
-                user: params.user,
-                url: url.href, // force toString to add a protocol to the URL.
-                method: request.method,
-                hash: bodyHash
-            });
-            request.addHeader({
-                key: 'Authorization',
-                value: result,
-                system: true
-            });
+        if (!params.includePayloadHash) {
```

Comment:
Add a brief description of why would someone opt-out or opt-in for payload hash.
And what's the consequence if we force include it for every time (existing behavior/bug)